source/Calamari/Integration/Nginx/NginxServer.cs

```diff
@@ -0,0 +1,287 @@
+
+                    AddServerBindingDirective(NginxDirectives.Server.CertificateKey,
+                        certificateKeyPath ?? binding.CertificateKeyLocation);
+
+                    var securityProtocols = binding.SecurityProtocols;
+                    if (securityProtocols != null && securityProtocols.Any())
+                    {
+                        AddServerBindingDirective(NginxDirectives.Server.SecurityProtocols, string.Join(" ", binding.SecurityProtocols));
+                    }
+
+//                    if (!string.IsNullOrWhiteSpace((string) binding.ciphers))
+//                    {
+//                        AddServerBindingDirective(NginxDirectives.Server.SslCiphers, string.Join(":", binding.ciphers));
+//                    }
+                }
+            }
+
+            return this;
+        }
+
+        public NginxServer WithHostName(string serverHostName)
+        {
+            if (string.IsNullOrWhiteSpace(serverHostName) || serverHostName.Equals("*")) return this;
+
+            useHostName = true;
+            hostName = serverHostName;
+
+            return this;
+        }
+
+        public NginxServer WithAdditionalLocations(IEnumerable<Location> locations)
+        {
+            if (!locations.Any()) return this;
+
+            var locationIndex = 0;
+            foreach (var location in locations)
+            {
+                var locationConfig = GetLocationConfig(location);
+                var sanitizedLocationName = SanitizeLocationName(location.Path, locationIndex.ToString());
+                var locationConfFile = Path.Combine(TempConfigRootDirectory, $"{virtualServerName}.conf.d",
+                    $"location.{sanitizedLocationName}.conf");
+
+                additionalLocations.Add(locationConfFile, locationConfig);
+                locationIndex++;
+            }
+
+            return this;
+        }
+
+        private string SanitizeLocationName(string locationPath, string defaultValue)
+        {
+            var match = Regex.Match(locationPath, "[a-zA-Z0-9/]+");
+            return match.Success 
+                ? match.Value.Replace("/", "_").Trim('_') 
+                : defaultValue;
+        }
+
+        public NginxServer WithRootLocation(Location location)
+        {
+            rootLocation = location;
+
+            return this;
+        }
+
+        public void BuildConfiguration()
+        {
+            virtualServerConfig =
+                $@"
+server {{
+{string.Join(Environment.NewLine, serverBindingDirectives.Select(binding => $"    {binding.Key} {binding.Value};"))}
+{(useHostName ? $"    {NginxDirectives.Server.HostName} {hostName};" : "")}
+{(additionalLocations.Any() ? $"    {NginxDirectives.Include} {GetConfigRootDirectory()}/{virtualServerName}.conf.d/location.*.conf;" : "")}
+{GetLocationConfig(rootLocation)}
+}}
+";
+        }
+
+        public void SaveConfiguration(string tempDirectory)
```

Comment:
I think it would be nice to see some tests that show sample configurations that could be output from this.